internal/core/thirdparty/tantivy/tantivy-binding/src/index.rs

```diff
@@ -0,0 +1,329 @@
+use std::fmt;
+use std::ops::Bound;
+use std::str::FromStr;
+
+use futures::executor::block_on;
+
+use tantivy::collector::TopDocs;
+use tantivy::directory::MmapDirectory;
+use tantivy::query::{Query, RangeQuery, TermQuery, RegexQuery};
+use tantivy::schema::{Field, IndexRecordOption, Schema, TextFieldIndexing, TextOptions, INDEXED};
+use tantivy::{doc, tokenizer, DocAddress, Index, IndexReader, IndexWriter, ReloadPolicy, Term};
+
+use crate::data_type::TantivyDataType;
+use crate::util::make_bounds;
+
+pub struct IndexWrapper {
+    pub field_name: Option<String>,
+    pub field: Option<Field>,
+    pub data_type: Option<TantivyDataType>,
+    pub path: Option<String>,
+    pub index: Option<Index>,
+    pub index_writer: Option<IndexWriter>,
+    pub reader: Option<IndexReader>,
+}
+
+impl IndexWrapper {
+    fn create_reader(&mut self) {
+        let reader = self
+            .index
+            .as_mut()
+            .unwrap()
```

Comment:
Too much unwrap. 
Remove Option in IndexWrapper definition and change create_reader as an associated function but not a method of IndexWrapper.

create_reader() -> IndexReader